### ISSUE TITLE
Add `rego.Data` function to simply add data from map

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/opa#5961]
Having to create a storage object, and possibly a transaction, in order to just add some data for evaluation using the Go API is a chore, and not intuitive unless you happen to know about it. It would be great if we added a `rego.Data` function that would allow passing a map to populate the `data` object.

(There's a longer story around issues with transactions and concurrency, but I'll save that for another issue)

**Comment by ashutosh-narkar** [comment]
The proposed `rego.Data` similar to  `rego.Load`, `rego.LoadBundle` and `rego.ParsedBundle` would require a transaction to be passed as well. Then should be just a write call on the store as part of preparing the query.

**Comment by anderseknert** [comment]
I'm using both `rego.Load` and `rego.ParsedBundle` without passing any transaction. I'm not sure I follow :)

**Comment by ashutosh-narkar** [comment]
I was referring to [this](https://sourcegraph.com/github.com/open-policy-agent/opa@db3b016f98763c277c625a3431f51da32c3eef78/-/blob/rego/rego.go?L969) comment where you set the store and that this would apply to `rego.Data` as well.

**Comment by anderseknert** [comment]
Ah, I see. Yes, that'd be the same, but the most important point here is that you shouldn't need to set the store or transactions unless you have special requirements that necessitate that, and just throwing some random data in for evaluation should not be one.

**Comment by ashutosh-narkar** [comment]
But then this would make it different than `rego.ParsedBundle` for instance since in the `rego.ParsedBundle` case you'd actually write the bundle contents to the store. But that's not how you imagine `rego.Data` working, correct?

**Comment by anderseknert** [comment]
No, I want it to be written to the store, just like when using those other helper functions 🙂
